mmpose/evaluation/metrics/hand_metric.py

```diff
@@ -0,0 +1,200 @@
+                 collect_device: str = 'cpu',
+                 prefix: Optional[str] = None) -> None:
+        super().__init__(collect_device=collect_device, prefix=prefix)
+        for mode in modes:
+            if mode not in self.METRICS:
+                raise ValueError("`mode` should be 'MPJPE', 'MRRPE', or "
+                                 f"'Handedness_Acc', but got '{mode}'.")
+
+        self.modes = modes
+
+    def process(self, data_batch: Sequence[dict],
+                data_samples: Sequence[dict]) -> None:
+        """Process one batch of data samples and predictions. The processed
+        results should be stored in ``self.results``, which will be used to
+        compute the metrics when all batches have been processed.
+
+        Args:
+            data_batch (Sequence[dict]): A batch of data
+                from the dataloader.
+            data_samples (Sequence[dict]): A batch of outputs from
+                the model.
+        """
+        for data_sample in data_samples:
+            # predicted keypoints coordinates, [1, K, D]
+            pred_coords = data_sample['pred_instances']['keypoints']
+            _, K, _ = pred_coords.shape
+            pred_coords_cam = pred_coords.copy()
+            # ground truth data_info
+            gt = data_sample['gt_instances']
+            # ground truth keypoints coordinates, [1, K, D]
+            gt_coords = gt['keypoints_cam']
+
+            keypoints_cam = gt_coords.copy()
+            # ground truth keypoints_visible, [1, K, 1]
+            mask = gt['keypoints_visible'].astype(bool).reshape(1, -1)
+
+            pred_hand_type = data_sample['pred_instances']['hand_type']
+            gt_hand_type = data_sample['hand_type']
+            if pred_hand_type is None and 'Handedness_Acc' in self.modes:
+                raise KeyError('metric Handedness_Acc is not supported')
+
+            pred_root_depth = data_sample['pred_instances']['rel_root_depth']
+            if pred_root_depth is None and 'MRRPE' in self.modes:
+                raise KeyError('metric MRRPE is not supported')
+
+            abs_depth = data_sample['abs_depth']
+            focal = data_sample['focal']
+            principal_pt = data_sample['principal_pt']
+
+            result = {}
+
+            if 'MPJPE' in self.modes:
+                keypoints_cam[..., :21, :] -= keypoints_cam[..., 20, :]
+                keypoints_cam[..., 21:, :] -= keypoints_cam[..., 41, :]
+
+                pred_coords_cam[..., :21, 2] += abs_depth[0]
+                pred_coords_cam[..., 21:, 2] += abs_depth[1]
+                pred_coords_cam = pixel_to_camera(pred_coords_cam, focal[0],
+                                                  focal[1], principal_pt[0],
+                                                  principal_pt[1])
+
+                pred_coords_cam[..., :21, :] -= pred_coords_cam[..., 20, :]
+                pred_coords_cam[..., 21:, :] -= pred_coords_cam[..., 41, :]
```

Comment:
Please avoid assuming the number of keypoints is 21

Reply:
renamed metric to `InterHandMetric`